WalletWasabi.Gui/Controls/WalletExplorer/CoinListViewModel.cs

```diff
@@ -197,37 +177,11 @@ public GridLength CoinJoinStatusWidth
 		private bool? GetCheckBoxesSelectedState(Func<CoinViewModel, bool> coinFilterPredicate)
 		{
 			var coins = Coins.Where(coinFilterPredicate).ToArray();
-			bool IsAllSelected = true;
-			foreach (CoinViewModel coin in coins)
-			{
-				if (!coin.IsSelected)
-				{
-					IsAllSelected = false;
-					break;
-				}
-			}
 
-			bool IsAllDeselected = true;
-			foreach (CoinViewModel coin in coins)
-			{
-				if (coin.IsSelected)
-				{
-					IsAllDeselected = false;
-					break;
-				}
-			}
+			bool isAllSelected = coins.All(coin => coin.IsSelected);
+			bool isAllDeselected = coins.All(coin => !coin.IsSelected);
 
-			if (IsAllDeselected)
-			{
-				return false;
-			}
-
-			if (IsAllSelected)
-			{
-				return true;
-			}
-
-			return null;
+			return isAllSelected && !isAllDeselected;
```

Comment:
I think it's incorrect. I think the null value is the [o] state of the checkbox, so when not all selected [x] nor all deselected [ ], but I didn't test it.

Reply:
Correct. This change removed the null value which is what I  first confused with 'selected'. I have fixed this, but am still setting them all to false on load, looks better I think.